VDA-Verification-Base/test/proof.ts

```diff
@@ -0,0 +1,184 @@
+                    uniqueId: "12345678"
+                }
+                `]
+            )
+
+            const nonce = (await contract.getNonce(didWallet.address)).toNumber()
+            const msgWithNonce = ethers.utils.solidityPack(
+                ['bytes','uint256'],
+                [rawMsg, nonce]
+              )
+            const signature = keyring.sign(msgWithNonce)
+    
+            await contract.testRawStringData(didWallet.address, rawMsg, signature, proof)
+            const updatedNonce = (await contract.getNonce(didWallet.address)).toNumber()
+
+            assert.equal((nonce+1), updatedNonce, 'Nonce updated');
+        });
+    })
+    
+    describe("Proof test with DIDDocument & DIDClient", () => {
+        it("On-chain verification with DIDDocument", async () => {
+            const {
+                didwallet,
+                account,
+                client,
+                context,
+                CONTEXT_NAME
+            } = await initVerida()
+
+            const PRIVATE_KEY = didwallet.privateKey
+            const ADDRESS = didwallet.address
+
+            console.log("Getting DID client...")
+            const didClient = await account.getDidClient()
+            const did = await account.did()
+            console.log("Getting keyring....")
+            const keyring = await account.keyring(CONTEXT_NAME)
+
+            const doc = await didClient.get(did)
+            const data = doc.export()
+
+            // Log the full DID Document
+            // console.log(data)
+
+            // Master keys controlled by the DID
+            const keys = await keyring.getKeys()
+            const publicKeyHex = keys.signPublicKeyHex
+            
+            // Proof 
+            const proofString = `${ADDRESS}${keys.signPublicAddress}`.toLowerCase()
+            const privateKeyBuffer = new Uint8Array(Buffer.from(PRIVATE_KEY.slice(2), 'hex'))
+            const proof = EncryptionUtils.signData(proofString, privateKeyBuffer)
+
+            const contextHash = DIDDocument.generateContextHash(doc.id, CONTEXT_NAME)
+            const didDocumentVerificationMethod = data.verificationMethod?.find(item => {
+                return item.id.match(`${doc.id}\\?context=${contextHash}&type=sign`)
+            })
+            // @ts-ignore
+            const didDocumentContextProof = didDocumentVerificationMethod.proof
+
+            // Verify the proof stored in the DID document for this context matches the expected
+            // proof that we generated above
+            assert.equal(proof, didDocumentContextProof)
+
+            const orgNonce = (await contract.getNonce(ADDRESS)).toNumber()
+            const params = ethers.utils.solidityPack(
+                ['string', 'uint'],
+                ['Test data', orgNonce]
+            )
+            const signature = await keyring.sign(params)
+
+            await contract.verifyRequestWithArray(ADDRESS, [ADDRESS], params, signature, didDocumentContextProof);
```

Comment:
Should get the response and assert it is okay

Reply:
See line number 180. It checks whether the nonce updated.